akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -514,37 +514,36 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
    * Read a number of elements from the given inlet and continue with the given function,
    * suspending execution if necessary. This action replaces the [[InHandler]]
    * for the given inlet if suspension is needed and reinstalls the current
-   * handler upon receiving the last `onPush()` signal (before invoking the `andThen` function).
+   * handler upon receiving the last `onPush()` signal.
+   *
+   * If upstream closes before N elements have been read,
+   * the `onClose` function is invoked with the elements which were read.
    */
   final protected def readN[T](in: Inlet[T], n: Int)(andThen: Seq[T] ⇒ Unit, onClose: Seq[T] ⇒ Unit): Unit =
+    //FIXME `onClose` is a poor name for `onComplete` rename this at the earliest possible opportunity
```

Comment:
Technically we could rename right now, and use `@deprecatedName` to keep the old bin compat